dll/cpl/sysdm/virtmem.c

```diff
@@ -569,27 +576,46 @@ OnSelChange(HWND hwndDlg, PVIRTMEM pVirtMem)
         }
 
         /* Set minimum pagefile size */
-        SetDlgItemText(hwndDlg, IDC_MINIMUM, _T("16 MB"));
+        SetDlgItemText(hwndDlg, IDC_MINIMUM, _T("2 MB"));
 
         /* Set recommended pagefile size */
         MemoryStatus.dwLength = sizeof(MEMORYSTATUSEX);
         if (GlobalMemoryStatusEx(&MemoryStatus))
         {
             FreeMemMb = (UINT)(MemoryStatus.ullTotalPhys / (1024 * 1024));
-            _stprintf(szBuffer, _T("%u MB"), FreeMemMb + (FreeMemMb / 2));
+            RecoMemMb = FreeMemMb + (FreeMemMb / 2);
+            if (RecoMemMb > 4096)
+                RecoMemMb = 4096;
+            _stprintf(szBuffer, _T("%u MB"), RecoMemMb);
             SetDlgItemText(hwndDlg, IDC_RECOMMENDED, szBuffer);
         }
 
         /* Set current pagefile size */
-#if 0
         PageFileSizeMb = 0;
-        for (i = 0; i < 26; i++)
+        		
+        for (i = 0; i < pVirtMem->Count; i++)
         {
-            PageFileSizeMb += pVirtMem->Pagefile[i].InitialSize;
+		    _stprintf(szText,
+                      _T("%c:\\pagefile.sys"),
+                      pVirtMem->Pagefile[i].szDrive[0]);
+		
+            hFind = FindFirstFileW(szText, &fdata);
+            if (hFind == INVALID_HANDLE_VALUE) 
+            {
+                DPRINT1("Unable to read PageFile size : %ls due to error %d\n", szText,GetLastError());				
+            }
+            else
+            {
+                ULONGLONG Size = fdata.nFileSizeLow;
```

Comment:
Use `ULARGE_INTEGER` for 64-bit integer handling.

Reply:
OK

Reply:
Cause compilation error :
../dll/cpl/sysdm/virtmem.c:609:17: error: invalid initializer
../dll/cpl/sysdm/virtmem.c:610:22: error: invalid operands to binary | (have 'ULARGE_INTEGER' and 'long long int')
../dll/cpl/sysdm/virtmem.c:611:29: error: invalid operands to binary / (have 'ULARGE_INTEGER' and 'int')
../dll/cpl/sysdm/virtmem.c:612:32: error: invalid operands to binary + (have 'UINT' and 'ULARGE_INTEGER')